Parse/Internal/File/State/PFFileState.m

```diff
@@ -78,13 +78,11 @@ - (NSString *)secureURLString {
     }
 
     NSString *scheme = components.scheme;
-    if (![scheme isEqualToString:@"http"]) {
-        return self.urlString;
-    }
-
-    if ([components.host isEqualToString:_PFFileStateSecureDomain]) {
+    if ([scheme isEqualToString:@"http"] || [components.host isEqualToString:_PFFileStateSecureDomain]) {
```

Comment:
Since this is internal API, it does indeed need some clarification from our end:

- `secureURLString` is the one that we are trying to use at all times by default
- `urlString` is the server-provided default URL
- Since we only know that we can automatically update `http` to `https` for parse.com S3 bucket - there is a hard-coded check on that domain
- For every other domain - whatever is sent from the server is the source of truth and we are not going to change it, since the server dictates where the files should be downloaded from.